addons/hr_attendance/controllers/main.py

```diff
@@ -61,6 +62,17 @@ def kiosk_menu_item_action(self):
         else:
             return request.not_found()
 
+    @http.route('/hr_attendance/kiosk_mode_menu/<int:company_id>', auth='user', type='http')
+    def kiosk_menu_item_action2(self, company_id):
+        if request.env.user.user_has_groups("hr_attendance.group_hr_attendance_manager"):
+            # Auto log out will prevent users from forgetting to log out of their session
+            # before leaving the kiosk mode open to the public. This is a prevention security
+            # measure.
+            request.session.logout(keep_db=True)
+            return request.redirect(request.env['res.company'].browse(company_id).attendance_kiosk_url)
+        else:
+            return request.not_found()
```

Comment:
alternatively you could do:

```suggestion
        request.update_context(allowed_company_ids=[company_id])
        return self.kiosk_menu_item_action()
```
and in the original method do:

```diff
-            return request.redirect(request.env.user.company_id.attendance_kiosk_url)
+            return request.redirect(request.env.company.attendance_kiosk_url)
```
this way the change is a little smaller and in master you can just modify the existing method